flaml/autogen/agent/human_proxy_agent.py

```diff
@@ -0,0 +1,121 @@
+        elif lang == "python":
+            if code.startswith("# filename: "):
+                filename = code[11 : code.find("\n")].strip()
+            else:
+                filename = None
+            exitcode, logs = execute_code(code, work_dir=self._work_dir, filename=filename)
+        else:
+            # TODO: could this happen?
+            exitcode, logs = 1, "unknown language"
+            # raise NotImplementedError
+        return exitcode, logs
+
+    def receive(self, message, sender):
+        """Receive a message from the sender agent.
+        Once a message is received, this function sends a reply to the sender or simply stop.
+        The reply can be generated automatically or entered manually by a human.
+        """
+        super().receive(message, sender)
+        # default reply is empty (i.e., no reply, in this case we will try to generate auto reply)
+        reply = ""
+        if self._human_input_mode == "ALWAYS":
+            code, lang = extract_code(message)
+            msg2display = (
+                f"\nCode block detected: \n{code}\nUse auto-reply to execute the code and return the result.\n"
+                if lang != "unknown"
+                else ""
+            )
+            reply = input(
+                "*" * 40
+                + f"{msg2display}Provide feedback to the sender. Press enter to skip and use auto-reply, or type 'exit' to end the conversation: "
+            )
+        elif self._consecutive_auto_reply_counter[
+            sender.name
+        ] >= self._max_consecutive_auto_reply or self._is_termination_msg(message):
+            if self._human_input_mode == "TERMINATE":
+                reply = input(
+                    "Please give feedback to the sender. (Press enter or type 'exit' to stop the conversation): "
+                )
+                reply = reply if reply else "exit"
+            else:
+                # this corresponds to the case when self._human_input_mode == "NEVER"
+                reply = "exit"
+        if reply == "exit" or (self._is_termination_msg(message) and not reply):
+            return
+        elif reply:
+            # reset the consecutive_auto_reply_counter
+            self._consecutive_auto_reply_counter[sender.name] = 0
+            self._send(reply, sender)
+            return
+
+        self._consecutive_auto_reply_counter[sender.name] += 1
+        if self._human_input_mode != "ALWAYS":
+            code, lang = extract_code(message)
+        if lang == "unknown":
+            # no code block is found, lang should be "unknown"
+            self._send(reply, sender)
+        else:
+            # try to execute the code
+            exitcode, logs = self._execute_code(code, lang)
+            exitcode2str = "execution succeeded" if exitcode == 0 else "execution failed"
+            self._send(f"exitcode: {exitcode} ({exitcode2str})\nCode output: {logs.decode('utf-8')}", sender)
```

Comment:
We can put these in a method `auto_reply()` so that one can override it later.

Reply:
Thanks! I revised it accordingly. Also revised the rocket plotting example in `test_agent.py`